hydrus/hydraspec/openapi-hydra_parser.py

```diff
@@ -0,0 +1,113 @@
+"""
+Module to take in Open Api Specification and convert it to HYDRA Api Doc
+
+"""
+import yaml
+from doc_writer import HydraDoc, HydraClass, HydraClassProp, HydraClassOp
+import json
+
+with open("openapi_sample.yaml", 'r') as stream:
+    try:
+        doc = yaml.load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+info = doc["info"]
+desc = info["description"]
+title = info["title"]
+baseURL = doc["host"]
+name = doc["basePath"]
+api_doc = HydraDoc(name, title, desc, name, baseURL)
+definitions = doc["definitions"]
+paths = doc["paths"]
+definitionSet = set()
+classAndClassDefinition = dict()
+hydra_doc = ""
+
+
+def getClasses():
+    """
+    Get Definitions from Open Api specification and convert to Classes and supported Props for Hydra Documentation
+    """
+    for class_ in definitions:
+        try:
+            desc = definitions[class_]["description"]
+        except KeyError:
+            desc = class_
+        classDefinition = HydraClass(class_, class_, desc, endpoint=False)
+        properties = definitions[class_]["properties"]
+        classAndClassDefinition[class_] = classDefinition
+        definitionSet.add(class_)
+        api_doc.add_supported_class(classDefinition, collection=False)
+        for prop in properties:
+            new_prop = HydraClassProp("vocab:" + prop, prop, required=False, read=True, write=True)
+            classAndClassDefinition[class_].add_supported_prop(new_prop)
+
+    get_ops()
+
+
+def generateEntrypoint():
+    """
+    Generates Entrypoint , Base Collection and Base Resource for the documentation
+    """
+    api_doc.add_baseCollection()
+    api_doc.add_baseResource()
+    api_doc.gen_EntryPoint()
+
+
+def get_ops():
+    """
+    Parses Paths from the Open Api Spec and creates supported operations for the paths which we have defined classes for
+    we check if the path has a class defined , if it does we parse the methods and add the information parsed to Hydra
+
+    :return:
+    """
+    for path in paths:
+        possiblePath = path.split('/')[1]
+        # dirty hack , do  case insensitive search more gracefully
+        possiblePath = possiblePath.replace(possiblePath[0], possiblePath[0].upper())
+        # check if the path name exists in the classes defined
+        if possiblePath in definitionSet and len(path.split('/')) == 2:
+            for method in paths[path]:
+                op_name = ""
+                op_method = method
+                op_expects = ""
+                op_returns = None
+                op_status = [{"statusCode": 200, "description": "dummyClass updated"}]
+                try:
+                    op_name = paths[path][method]["summary"]
+                except KeyError:
+                    op_name = possiblePath
+                try:
+                    parameters = paths[path][method]["parameters"]
+                    for param in parameters:
+                        op_expects = param["schema"]["$ref"].split('/')[2]
```

Comment:
Should be `'vocab:'+param["schema"]["$ref"].split('/')[2]`. ALso what if the parameter is something as simple as an integer and it's not defined in schema?

For example -  
``` yaml
      parameters:
        - name: petId
          in: path
          description: ID of pet to return
          required: true
          type: integer
          format: int64
```

Reply:
I have planned on adding some datatypes which Open api support like string , integer etc as classes and add those when we encounter them . I am adding the "vocab" part later in line 94